parser_library/src/workspaces/file_manager.h

```diff
@@ -61,11 +61,18 @@ class file_manager : public virtual diagnosable
     // Returns nullptr if there is no such file.
     virtual processor_file_ptr find_processor_file(const file_location& key) = 0;
 
-    // Returns list of all files in a directory. Returns associative array with pairs file location - file name.
-    virtual list_directory_result list_directory_files(const utils::resource::resource_location& directory) = 0;
-
-    virtual bool file_exists(const std::string& file_name) = 0;
-    virtual bool lib_file_exists(const std::string& lib_path, const std::string& file_name) = 0;
+    // Returns list of all files in a directory. Returns associative array with pairs file name - file location.
+    virtual list_directory_result list_directory_files(const utils::resource::resource_location& directory) const = 0;
+
+    // Returns list of all sub directories and symbolic links. Returns associative array with pairs file path/uri - file
+    // location.
+    virtual list_directory_result list_directory_subdirs_and_symlinks(
```

Comment:
Basically, yes. 

I can remove this together with `list_directory_files()` and use `content_loader` directly. That would mean minor refactoring of the `workspace.cpp` as the tests would need to have the possibility to provide a mock of `content_loader` instead of the real thing. Do you want to go in this direction?